test/index.ts

```diff
@@ -1,19 +1,52 @@
 import { expect } from "chai";
 import { ethers } from "hardhat";
 
-describe("Greeter", function () {
-  it("Should return the new greeting once it's changed", async function () {
-    const Greeter = await ethers.getContractFactory("Greeter");
-    const greeter = await Greeter.deploy("Hello, world!");
-    await greeter.deployed();
+describe("BarterMarket", function () {
+  before(async function () {
+    const [owner, account1, account2] = await ethers.getSigners();
 
-    expect(await greeter.greet()).to.equal("Hello, world!");
+    // Deploy Mock ERC20 Contract
+    const ERC20Factory = await ethers.getContractFactory("MockERC20");
+    const ERC20 = await ERC20Factory.deploy("Mock ERC20", "MERC20", 50000);
+    await ERC20.deployed();
 
-    const setGreetingTx = await greeter.setGreeting("Hola, mundo!");
+    console.log("ERC20 Deployed", ERC20.address);
 
-    // wait until the transaction is mined
-    await setGreetingTx.wait();
+    // Deploy Mock ERC721 Contract
+    const ERC721Factory = await ethers.getContractFactory("MockERC721");
+    const ERC721 = await ERC721Factory.deploy("Mock ERC721", "MERC721");
+    await ERC721.deployed();
 
-    expect(await greeter.greet()).to.equal("Hola, mundo!");
+    console.log("ERC721 Deployed", ERC721.address);
+
+    // Transfer ERC20 tokens to accounts
+    await ERC20.transfer(account1.address, 1000);
+    await ERC20.transfer(account2.address, 2000);
+
+    // Mint NFTs
+
+    // TokenIDs: 1 - 5
+    for (var i = 0; i < 5; i++) {
+      await ERC721.connect(account1).mint();
+    }
+
+    // TokenIDs: 6 - 15
+    for (var i = 0; i < 10; i++) {
+      await ERC721.connect(account2).mint();
+    }
+
```

Comment:
I'm not sure if this is happens when the signers are created, but it would probably be useful to give each account some eth

Reply:
Hardhat adds a default balance to these accounts :)